canon/src/store/mod.rs

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use cfg_if::cfg_if;
+
+use core::fmt;
+
+use crate::id::IdHash;
+use crate::CanonError;
+
+cfg_if! {
+    if #[cfg(feature = "host")] {
+        mod host;
+        use host::HostStore as Inner;
+    } else if #[cfg(feature = "bridge")] {
+        mod bridge;
+        use bridge::BridgeStore as Inner;
+    } else {
+        mod void;
+        use void::VoidStore as Inner;
```

Comment:
Maybe we can drop few lines of comments here: for example, what is the `VoidStore`, why is needed and when it will be used?

I assumed it's needed because we need some placeholder when neither `host` or `hosted` / `bridge` is selected, but it looks to me a workaround / patch. Maybe we can have a quick call and discuss about it.

Reply:
Added comments in void.rs